src/lambdas/RedriveSNSDQLEvents/writeToFirehose.test.ts

```diff
@@ -0,0 +1,142 @@
+import { PutRecordBatchCommandOutput } from '@aws-sdk/client-firehose'
+import { when } from 'jest-when'
+import { firehosePutRecordBatch } from '../../sharedServices/firehose/firehosePutRecordBatch'
+import { logger } from '../../sharedServices/logger'
+import { auditEventsToFirehoseRecords } from '../../utils/helpers/firehose/auditEventsToFirehoseRecords'
+import {
+  allSuccessFirehoseResponseExpectedResult,
+  baseFirehoseResponse,
+  baseProcessingResults,
+  mockFireHoseRecords
+} from '../../utils/tests/test-helpers/redriveSNSDLQTestHelper'
+import * as firehoseFunctions from './writeToFirehose'
+import { FirehoseProcessingResult, writeToFirehose } from './writeToFirehose'
+
+jest.mock('../../sharedServices/firehose/firehosePutRecordBatch', () => ({
+  firehosePutRecordBatch: jest.fn()
+}))
+
+jest.mock(
+  '../../utils/helpers/firehose/auditEventsToFirehoseRecords.ts',
+  () => ({
+    auditEventsToFirehoseRecords: jest.fn()
+  })
+)
+
+describe('test parseFirehoseResponse() function', () => {
```

Comment:
separating `parseFirehoseResponse` and `writeToFirehose` might also make importing and mocking the respective function a bit easier as they would have separate test files as well

Reply:
sorted